commons/sync/timedmutex.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.package rpcutils
+
+package sync
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+// A TimedLocker represents an object that can be locked and unlocked.
+type TimedLocker interface {
+	Lock(string)
```

Comment:
I guess there's no way to embed the Locker interface here, because of the addition of a Lock(string) method.